pkg/asset/manifests/powervs/cluster.go

```diff
@@ -56,11 +55,17 @@ func GenerateClusterAssets(installConfig *installconfig.InstallConfig, clusterID
 
 	network = fmt.Sprintf("%s-network", clusterID.InfraID)
 
-	n, err := rand.Int(rand.Reader, big.NewInt(253))
-	if err != nil {
-		return nil, fmt.Errorf("failed to generate random subnet: %w", err)
+	logrus.Debugf("GenerateClusterAssets: len MachineNetwork = %d", len(installConfig.Config.Networking.MachineNetwork))
+	switch len(installConfig.Config.Networking.MachineNetwork) {
+	case 0:
+		dhcpSubnet = powervsdefaults.DefaultMachineCIDR.String()
+	case 1:
+		dhcpSubnet = installConfig.Config.Networking.MachineNetwork[0].CIDR.String()
+	default:
+		logrus.Infof("Warning: multiple machineNetwork found! Ignoring.")
+		dhcpSubnet = powervsdefaults.DefaultMachineCIDR.String()
```

Comment:
`installConfig.Config.Networking.MachineNetwork` should always be set at this point, either with the user's value or with the default value:
```suggestion
	dhcpSubnet = installConfig.Config.Networking.MachineNetwork[0].CIDR.String()
	if numNetworks := len(installConfig.Config.Networking.MachineNetwork); numNetworks > 1 {
	    logrus.Infof("Warning: %d machineNetwork found! Ignoring all except first.", numNetworks)
	}
```
Also if users specify more than one CIDR block, do we really want to ignore all and use our hardcoded default? Can't we at least use the first value?